test/mysql2_test.rb

```diff
@@ -160,6 +160,26 @@ def test_resource_acquisition_for_query
     end
   end
 
+  def test_semian_allows_rollback
+    client = connect_to_mysql!
+
+    client.query('START TRANSACTION;')
+
+    Semian[:mysql_testing].acquire do
+      client.query('ROLLBACK;')
+    end
+  end
```

Comment:
This Semian resource has one ticket. Nothing else has claimed it during this block as far as I can see, so I'm confused.